main/src/main/java/cgeo/geocaching/maps/mapsforge/v6/layers/PositionLayer.java

```diff
@@ -77,7 +77,7 @@ public void draw(final BoundingBox boundingBox, final byte zoomLevel, final Canv
             arrowNative = ImageUtils.convertToBitmap(ResourcesCompat.getDrawable(CgeoApplication.getInstance().getResources(), R.drawable.my_location_chevron, null));
             rotateArrow();
         }
-        if (arrow != null) {
```

Comment:
I suggest an alternative approach:
- before line 89, store arrow in a local variable (e.g. `final Bitmap lArrow = this.arrow;`
- then use this local variable instead (e.g. line 80 would read `if (lArrow != null) {`

That way, race conditions would not be able to set lArrow to null in-between it's usage.

Reply:
Thanks for the proposal. How would this approach compare resource & performance-wise?

Reply:
> Thanks for the proposal. How would this approach compare resource & performance-wise?

I don't think resource/performance is really an impact in this case either way, since the majority of CPU in this class is probably used by recalculating rotation of the image over and over again. Compared to this, everything we do here is probably neglectable.

But if we go into detail analysis, I would assume that the approach with local var is more performant. It doesn't create any new objects (just a pointer) and the arrow bitmap might be hold into memory a few milliseconds longer. On the other hand you save the creation and catching of an exception, and creation of exceptions is expensive in terms of performance. Which is ok since they are exceptions and should not be used for normal control flow of application logic. 

The other aspect is that you should normally not catch runtime exceptions like NPE anyway. If necessary to fix urgent production issues that's ok imho, but it should only ever be a last resort. Because if code generates NPEs then something is not right with it.

Reply:
Thanks for the detailed explanation. Just for my understanding (and sorry if this might be a dumb question):

It looks to me that several instances of `draw()` get called in parallel, thus leading to race conditions if one instance has done the `if (arrow != null)` check already, but not yet the actual drawing, and before that the other instance overwrites/deletes `arrow`, thus the first instance runs into the NPE when trying to do their calculation or the actual drawing.

Now, if the first instance creates a pointer to `arrow`, then the other instance modifies/deletes `arrow`, what will the pointer of the first instance point to? Will it create a local copy ("copy on modify") before the other instance modifies the object, or will it still point to the same memory space the other instance is modifying?

Reply:
All variables in Java store references ("pointers") with the exception of primitive types. In our case this would mean that
* Start point is that `this.arrow` points to a valid `Bitmap` containing the arrow image
* One thread calls `draw()`, the reference to this bitmap is stored in local `lArrow` variable
* Another thread somehow sets `this.arrow` to `null`. Which means `this.arrow` doesn't point to the bitmap any more. If this would have been the last reference to the bitmap, then Garbage Collector would kick in and delete the bitmap eventually. But in our case we would still have our local variable `lArrow` pointing to it so GC leaves it in memory.
* Now our first thread is doing the drawing. The method is finished at some point which makes our local variable `lArrow` invalid. If now this was the last reference to the actual bitmap, then at this point GC would delete it from memory

So there will not be any "implicit" copying of the actual bitmap, it is just "hold" into memory a few more millisceonds

Having said that: I just looked into code, and it really really puzzles me how this.arrow can ever become null again after it was non-null for the first time. Because the only place where something is assigned to arrow is line 106, and there another object (not null) is assigned to it...:

`arrow = AndroidGraphicFactory.convertToBitmap(tmpArrow);`

Reply:
Thanks for the detailed explanation. I'm always learning something new...
Have adapted the PR accordingly.